main.py

```diff
@@ -142,6 +142,16 @@ def initSnakeLadder(self):
                 height = randint(1, 3)
                 x1, y1 = i, randint(height, self.numRows - 1)
                 x2, y2 = i, y1 - height
+
+                # There is a small change that a snake may be placed starting at the finishing square
+                # If this happens, we can move the starting point down by 1
+                if (self.numRows % 2 == 1 and x1 == self.numCols - 1 and y1 == self.numRows - 1) or (self.numRows % 2 == 0 and x1 == 0 and y1 == self.numRows - 1):
```

Comment:
This is a very messy statement, but I couldn't see an easy way to simplify it.

There are 2 cases:

 - Case 1 (number of rows is even)
   - In this case we need to check the top right square 
 - Case 2 (number of rows is odd)
   - In this case we need to check to top left square